### PR TITLE
Fix the endianness issue in v1 frozen graphs in python:lite_test on BE machines

### PR DESCRIPTION
**Pull request by kun-lu20**
This PR is to add big-endian support to TensorFlow v1 frozen graphs.

Sub-tests `FromFrozenGraphObjectDetection.testTFLiteGraphDef` and `FromFrozenGraphObjectDetection.testModifyIOToUint8` in `//tensorflow/lite/python:lite_test` fail on s390x (big-endian (BE) arch) because the tensor content in `graph_def` which is parsed from the file containing frozen GraphDef is in little-endian (LE) format.
 
To solve this issue, this PR refactored the `byte_swap_tensor` related code in `tensorflow/python/saved_model/utils_impl.py` and added helper function `swap_tensor_content_in_graph_node()` to deal with the LE/BE conversion in frozen graphs. This function is invoked in frozen graph read/write process so that v1 frozen graphs would be stored with LE format in files and converted to BE format right after being loaded into memory on BE machines.
 
This code change also extended the covered code paths for byte-swapping in read/write process of TensorFlow v2 saved models.
 
This PR could fix the above mentioned sub-tests in `//tensorflow/lite/python:lite_test` on s390x (BE machines) and will not cause any regressions on LE/BE platforms.

Fixes issue https://github.com/tensorflow/tensorflow/issues/57215 .

Signed-off-by: Kun-Lu <kun.lu@ibm.com>

### CODE REVIEW

**Comment by kun-lu20** [comment]
Hi @alankelly ,

Could you please review this PR when you have some time? Thank you very much!

**Comment by kun-lu20** [comment]
Hi @alankelly ,

Hope all is well. 

Could you please have a look at this PR? Thank you!

**Comment by alankelly** [comment]
@terryheo Can you PTAL at this? This is really not my domain. Thanks1

**Comment by kun-lu20** [comment]
Hi @terryheo ,

Could you please review this PR when you have some time? Thank you very much!

File: tensorflow/python/framework/byte_swap_tensor.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+def byte_swap_tensor_content(tensor, from_endiness, to_endiness):
+  """Byte swaps."""
+  if tensor.dtype in byte_swappable:
+    tshape = tensor.tensor_shape.dim
+    tensor_bytes = tensor.tensor_content
+    if tensor_bytes:
+      tensor_size = 1
+      for sz in tshape:
+        tensor_size = tensor_size * sz.size
```
Comment (terryheo) [inline tensorflow/python/framework/byte_swap_tensor.py:39]: sz.size is always bigger than zero?

File: tensorflow/python/framework/graph_io.py [hunk 2]
```diff
@@ -58,6 +60,14 @@ def write_graph(graph_or_graph_def, logdir, name, as_text=True):
   else:
     graph_def = graph_or_graph_def
 
+  if sys.byteorder == 'big':
+    if hasattr(graph_def, 'node'):
```
Comment (terryheo) [inline tensorflow/python/framework/graph_io.py:64]: Is this condition guaranteed only for frozen graph?

File: tensorflow/python/framework/byte_swap_tensor.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+# Based on tensor_bundle/byte_swap.cc
+byte_swappable = [
+    dtypes.float16, dtypes.float32, dtypes.float64, dtypes.bfloat16,
+    dtypes.complex64, dtypes.complex128, dtypes.uint16, dtypes.uint32,
+    dtypes.uint64, dtypes.int16, dtypes.int32, dtypes.int64, dtypes.qint16,
+    dtypes.quint16, dtypes.qint32
+]
+
+
+def byte_swap_tensor_content(tensor, from_endiness, to_endiness):
```
Comment (terryheo) [inline tensorflow/python/framework/byte_swap_tensor.py:31]: Can we have tests for these methods?

**Review by terryheo — COMMENTED** [review]
Actually changes are for TF framework. We need an additional review from TF framework team.

File: tensorflow/python/framework/byte_swap_tensor.py (hunk 1 above)
Comment (kun-lu20) [reply tensorflow/python/framework/byte_swap_tensor.py]: A guarded condition has been added to test if sz.size is not equal to zero. Thanks!

File: tensorflow/python/framework/graph_io.py (hunk 2 above)
Comment (kun-lu20) [reply tensorflow/python/framework/graph_io.py:64]: No, having attribute `node` can't guarantee it is a frozen graph. I've refactored the `swap_tensor_content_*` function names accordingly. Thanks!

File: tensorflow/python/framework/byte_swap_tensor.py (hunk 3 above)
Comment (kun-lu20) [reply tensorflow/python/framework/byte_swap_tensor.py:31]: These methods will only be invoked if the host arch is big-endian.

The following existing test cases could verify the functionality of these methods on BE machines:
```
//tensorflow/python/distribute/integration_test:saved_model_test_cpu
//tensorflow/python/saved_model:load_v1_in_v2_test
//tensorflow/python/saved_model:loader_test
//tensorflow/python/saved_model:save_test
//tensorflow/python/saved_model:saved_model_test
//tensorflow/python/saved_model:utils_test
//tensorflow/python/tools:saved_model_utils_test
//tensorflow/python/saved_model:load_test_cpu
//tensorflow/python/training:saver_test_cpu
//tensorflow/python/training:saver_test_2gpu
//tensorflow/python/data/kernel_tests:interleave_test
//tensorflow/lite/python:lite_test
```

**Comment by kun-lu20** [comment]
Hi @terryheo ,

Thanks for your valuable comments! I've improved the code change and updated the PR description accordingly. Could you please take a look?

**Comment by kun-lu20** [comment]
Hi @terryheo ,

Happy New Year!

Could you please review this PR again? Thank you very much!

**Comment by kun-lu20** [comment]
Hi @terryheo ,

I've solved the conflicts from the master branch. Could you please take a look at this PR? 

Thank you very much!

**Comment by kun-lu20** [comment]
Hi @terryheo ,

Could you please review this PR again when you have some time?

Thank you very much!

**Review by terryheo — APPROVED** [review]
LGTM

**Comment by kun-lu20** [comment]
Thanks @terryheo !

**Pull request merged**